src/main/java/org/broadinstitute/hellbender/tools/spark/ApplyBQSRSpark.java

```diff
@@ -48,15 +49,22 @@
     @Argument(doc = "If specified, shard the output bam", shortName = "shardedOutput", fullName = "shardedOutput", optional = true)
     private boolean shardedOutput = false;
 
+    @Argument(doc="The output parallelism, sets the number of reducers. Defaults to one partition per 10MB of input.",
+            shortName = "P", fullName = "parallelism", optional = true)
+    protected int parallelism = 0;
+
     @Override
     protected void runTool(JavaSparkContext ctx) {
         JavaRDD<GATKRead> initialReads = getReads();
+        if (parallelism == 0) {
+            parallelism = ReadsSparkSource.getRecommendedParallelism(ctx, getReadSourceName(), initialReads.partitions().size());
+        }
         final GCSOptions gcsOptions = getAuthenticatedGCSOptions(); // null if we have no api key
         Broadcast<RecalibrationReport> recalibrationReportBroadCast = ctx.broadcast(new RecalibrationReport(BucketUtils.openFile(bqsrRecalFile, gcsOptions)));
         final JavaRDD<GATKRead> recalibratedReads = ApplyBQSRSparkFn.apply(initialReads, recalibrationReportBroadCast, getHeaderForReads(), applyBQSRArgs);
 
         try {
-            ReadsSparkSink.writeReads(ctx, output, recalibratedReads, getHeaderForReads(), shardedOutput ? ReadsWriteFormat.SHARDED : ReadsWriteFormat.SINGLE);
+            ReadsSparkSink.writeReads(ctx, output, recalibratedReads, getHeaderForReads(), shardedOutput ? ReadsWriteFormat.SHARDED : ReadsWriteFormat.SINGLE, parallelism);
```

Comment:
If we move the `shardedOutput` and `parallelism` args up into `GATKSparkTool`, it would enable us to add a centralized `GATKSparkTool.writeReads()` method there as well that would handle the `shardedOutput` and `parallelism` settings internally, look simpler to client code, and help enforce consistency across tools. This makes sense since `GATKSparkTool.getReads()` already exists.

One design goal for GATK4 is to have the "engine" handle the low-level details of data handling as much as possible, to spare tool authors from having to add a lot of boilerplate to their code that they may not fully understand.